simde/x86/avx2.h

```diff
@@ -1569,31 +1569,15 @@ simde_mm256_hadd_epi16 (simde__m256i a, simde__m256i b) {
   return _mm256_hadd_epi16(a, b);
 #else
   simde__m256i_private r_;
-  simde__m256i_private a_ = simde__m256i_to_private(a);
-  simde__m256i_private b_ = simde__m256i_to_private(b);
 
-  r_.i16[0] = a_.i16[1] + a_.i16[0];
-  r_.i16[1] = a_.i16[3] + a_.i16[2];
-  r_.i16[2] = a_.i16[5] + a_.i16[4];
-  r_.i16[3] = a_.i16[7] + a_.i16[6];
-  r_.i16[4] = b_.i16[1] + b_.i16[0];
-  r_.i16[5] = b_.i16[3] + b_.i16[2];
-  r_.i16[6] = b_.i16[5] + b_.i16[4];
-  r_.i16[7] = b_.i16[7] + b_.i16[6];
-  r_.i16[8] = a_.i16[9] + a_.i16[8];
-  r_.i16[9] = a_.i16[11] + a_.i16[10];
-  r_.i16[10] = a_.i16[13] + a_.i16[12];
-  r_.i16[11] = a_.i16[15] + a_.i16[14];
-  r_.i16[12] = b_.i16[9] + b_.i16[8];
-  r_.i16[13] = b_.i16[11] + b_.i16[10];
-  r_.i16[14] = b_.i16[13] + b_.i16[12];
-  r_.i16[15] = b_.i16[15] + b_.i16[14];
+  r_.m128i[0] = simde_mm_add_epi16(simde_x_mm_deinterleaveeven_epi16(simde_mm256_extracti128_si256(a,0),simde_mm256_extracti128_si256(b,0)),simde_x_mm_deinterleaveodd_epi16(simde_mm256_extracti128_si256(a,0),simde_mm256_extracti128_si256(b,0)));
+  r_.m128i[1] = simde_mm_add_epi16(simde_x_mm_deinterleaveeven_epi16(simde_mm256_extracti128_si256(a,1),simde_mm256_extracti128_si256(b,1)),simde_x_mm_deinterleaveodd_epi16(simde_mm256_extracti128_si256(a,1),simde_mm256_extracti128_si256(b,1)));
```

Comment:
These are really a lot to cram into a single line.  You should break it up.

You can use the private members of `a_` and `b_` (which you would need to add back in) instead of `simde_mm256_extracti128_si256`.  Something like (completely untested, for illustrative purposes only):

```c
simde__m256i_private
  r_,
  a_ = simde__m256i_to_private(a),
  b_ = simde__m256i_to_private(b);

SIMDE_VECTORIZE
for (size_t = i = 0 ; i < (sizeof(r_.m128i) / sizeof(r_.m128i[0])) ; i++ ) {
  const simde__m128i even = simde_x_mm_deinterleaveeven_epi16(a_.m128i[i], b_.m128i[i]);
  const simde__m128i odd = simde_x_mm_deinterleaveodd_epi16(a_.m128i[i], b_.m128i[i]);
  r_.m128i[i] = simde_mm_add_epi16(even, odd);
}
```